### ISSUE TITLE
Process an Array of related entities

### ISSUE THREAD

**Issue by chucky2305** [nelmio/alice#968]
I am on Version dev-master of nelmio-alice right now and this is how to reproduce the bug.

```yaml
App\Entity\Group:
    group_one:
        __construct:
            name: 'Group 1'
   group_two:
        __construct:
            name: 'Group 2'


App\Entity\Person:
    person_admin:
        __construct:
            name: 'Admin'
            groups: '<collection([@group_one, @group_two])>'
```

```php
final class CollectionProvider extends BaseProvider
{
    /**
     * @param array $args
     * @return ArrayCollection
     */
    public function collection(array $args)
    {
        return new ArrayCollection($args);
    }
}
```

```php
class Person {

   private $name;

   /** @var Collection|Groups[] */
   private $groups;

    public function __construct(
        string $name,
        Collection $groups
    ) {
        $this->setName($name);
        $this->groups = $groups;
    }
}
```

**Comment by theofidry** [comment]
Is the bug still there when not using `collection`?

**Comment by chucky2305** [comment]
Forgot to mention the actual error message that comes up when i run my tests:
```
 Expected value of type "App\Entity\Zld" for association field "App\Entity\Stva#$zlds", got "Nelmio\Alice\Definition\Value\FixtureReferenceValue" instead. (Doctrine\ORM\ORMInvalidArgumentException)
```

**Comment by theofidry** [comment]
Err, what does `Stva#$zlds` has to do with the above?

**Comment by chucky2305** [comment]
I corrected the error message above

**Comment by theofidry** [comment]
Closed by #984
